src/data/validation.py

```
@@ -1,17 +1,60 @@
+            
+        try:
+            # The validate_email function checks for both syntax and MX records by default.
+            # check_deliverability=True is the default, but we're explicit here.
+            validate_email(email, check_deliverability=True)
+            return True
+        except EmailNotValidError as e:
+            # Log the reason for the invalid email for debugging purposes.
+            # Use logging.debug to avoid cluttering the console unless needed.
+            logging.debug(f"Invalid email address '{email}': {e}")
+            return False
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)